java/src/com/ibm/streamsx/rest/primitives/InstancesArray.java

```diff
@@ -0,0 +1,51 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2017
+ */
+package com.ibm.streamsx.rest.primitives;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.gson.Gson;
+import com.ibm.streamsx.rest.StreamsConnection;
+import com.ibm.streamsx.rest.primitives.Instance;
+import com.ibm.streamsx.rest.primitives.InstanceGson;
+
+public class InstancesArray {
```

Comment:
When querying for the list of instances, the json structure that comes back includes a an array of instances, a resource type and the total number of instances coming back.   GSON needed a structure that it could understand so that I didn't need to manipulate the JSON to get to the array.   

Looking at this, I can clean up the code in the array objects some more, and make them package level objects only, not public.